### DISCUSSION TITLE
Fastapi CORS error in Frontend

### DISCUSSION THREAD

**Discussion by bengabp** [deta/cloud-docs#473] [General]
I am getting CORS related errors when I try to call my fastapi api hosted on Deta from my javascript code.
Everything works fine when im accessing the endpoints from my browser
I have also added this code but it seems to not work
```
app = FastAPI()
app.add_middleware(
  CORSMiddleware,
  allow_origins=['*'],
  allow_credentials=True,
  allow_headers=['*'],
  allow_methods=["*"],
)
```

**Comment by jnsougata** [comment]
what's the error you are getting?

**Reply by bengabp** [reply]
Oh... sorry...I read a solution on github that said I should use the Cors Middleware from starlette
~~from fastapi.middleware.cors import CORSMiddleware~~
`from starlette.middleware.cors import CORSMiddleware`

**Reply by Sokhavuth** [reply]
According to the test conducted by experts in 2022-3-14 Balcksheep and Sanic are the fastest Python asynchronous web frameworks. We should try to use these two frameworks instead of FastAPI. Sanic web framework is very similar to Flask.
![picture](https://user-images.githubusercontent.com/49101948/189935206-59b63ba9-971d-41cc-b201-17c9172fdec0.png)

**Reply by jnsougata** [reply]
> According to the test conducted by experts in 2022-3-14 Balcksheep and Sanic are the fastest Python asynchronous web frameworks. We should try to use these two frameworks instead of FastAPI. Sanic web framework is very similar to Flask. ![picture](https://user-images.githubusercontent.com/49101948/189935206-59b63ba9-971d-41cc-b201-17c9172fdec0.png)

It doesn't matter if you use `asgi` or `wsgi` as long as you are using serverless platform. **The perf of your app will depend on the perf of the base route handler and middlewares.** Moreover this benchmark results are improper as `aiohttp` is way more faster than `sanic` and all.

**Reply by Wallvon** [reply]
> According to the test conducted by experts in 2022-3-14 Balcksheep and Sanic are the fastest Python asynchronous web frameworks. We should try to use these two frameworks instead of FastAPI. Sanic web framework is very similar to Flask. ![picture](https://user-images.githubusercontent.com/49101948/189935206-59b63ba9-971d-41cc-b201-17c9172fdec0.png)

That is for a traditional server, not a serverless platform. Besides, you should take those benchmarks with a grain of salt since https://web-frameworks-benchmark.netlify.app/result?l=python for example gets way different results and other benchmarks (or even different runs of the same benchmark) seem to also get different results. And, if you truly need to handle a lot of reqs/sec you wouldn't be using something like Python anyway since then something like Go would be more appealing.
